Servers/Administration/OJS.Servers.Administration/Controllers/ContestsController.cs

```diff
@@ -247,5 +251,10 @@ private async Task AddIpsToContest(Contest contest, string mergedIps)
                 await this.participantsData.InvalidateByContestAndIsOfficial(contest.Id, isOfficial: false);
             }
         }
+
+        private Expression<Func<Contest, bool>>? GetMasterGridFilter()
+            => c => c.LecturersInContests.Any(l => l.LecturerId == this.User.GetId()) ||
+                 c.Category!.LecturersInContestCategories.Any(lc => lc.LecturerId == this.User.GetId()) ||
+                 this.User.IsAdmin();
```

Comment:
Same here. This check better be first in order

Reply:
This is reworked in a business service